### DISCUSSION TITLE
Can this website be "installed" via any browser?

### DISCUSSION THREAD

**Discussion by HunterHeston** [coder/code-server#3902] [General] [answered]
I followed the instructions to set this up using a DigitalOcean server. I would love to "install" this site using the mechanism that Chrome provides(PWAs I believe). The same way that you can install something like google meet. 

Is that possible currently? I needed I can provide more detail.

**Comment by lg-nineoneninethreenine** [comment]
Already does. Except, it might be local servers or Railway deployments or both.

**Reply by HunterHeston** [reply]
Gotcha, I am running an instance on DigitalOcean following theses [instructions](https://github.com/cdr/deploy-code-server/blob/main/guides/digitalocean.md). That sounds like it may be the issue?

**Comment by jsjoeio** [comment, marked as answer]
To be clear, you're trying to install code-server via the PWA? 

After setting up on Digital Ocean:
1. Navigate to URL where code-server is running
2. Click the install PWA button in the URL bar
![image](https://user-images.githubusercontent.com/3806031/128404643-ddf4127a-f44a-4ae1-aeb4-8677acb0132a.png)
3. Use PWA 

Hope that helps!

**Reply by lg-nineoneninethreenine** [reply]
Exactly @jsjoeio that was what I was trying to tell.

**Reply by jsjoeio** [reply]
Go team! 🙌

**Reply by HunterHeston** [reply]
Yes that is exactly what I'm asking for! 

However it's not an option for me when using the xxxx-hunterheston.cdr.co domain.
<img width="1312" alt="Screen Shot 2021-08-06 at 7 57 17 PM" src="https://user-images.githubusercontent.com/21689157/128580935-618cdfc1-8e31-499b-800c-dee97075eecf.png">

Not sure if this has anything to do with connecting to a local instance vs a remote instance. Sorry if this is just an issue on my end.

I'm running: 
macOS: 11.0.1 
Chrome: 92.0.4515.131

**Reply by lg-nineoneninethreenine** [reply]
Well, idk about that but it works for deployed/local instances. Maybe not for a `.cdr.co` kind of domain. Even if it is paired to your local instance(chances)

**Reply by lg-nineoneninethreenine** [reply]
If you want a deployed instance, use Railway. It works, but slowly for me, I don't know the problem tho.

**Reply by jsjoeio** [reply]
Okay I tested with `--link` locally macOS + Chrome and it's working for me 🤔

https://user-images.githubusercontent.com/3806031/128784102-eefcc38c-8a60-4938-a3af-4b477aa47f08.mov

A couple thoughts: 
- it's being served on https, right?
- are there any logs in the console related to the service worker? it may be failing
- can you look to see if there are any errors reported in the manifest? use [this](https://stackoverflow.com/a/63524621/3015595)

**Reply by HunterHeston** [reply]
Alright got it working! It was the service worker failing to register after 404ing on the `serviceWorker.js` script. 

Here is the console output:

[Service Worker] registration: Failed to register a ServiceWorker for scope ('https://xxx-hunterheston.cdr.co/') with script ('https://xxx-hunterheston.cdr.co/static/4e8cd09ef0412dfc7b148b7639a692e20e4fd6dd/usr/lib/code-server/dist/serviceWorker.js'): A bad HTTP response code (404) was received when fetching the script. TypeError: Failed to register a ServiceWorker for scope ('https://xxx-hunterheston.cdr.co/') with script ('https://xxx-hunterheston.cdr.co/static/4e8cd09ef0412dfc7b148b7639a692e20e4fd6dd/usr/lib/code-server/dist/serviceWorker.js'): A bad HTTP response code (404) was received when fetching the script.

Confirmed that `usr/lib/code-server/dist/serviceWorker.js` does not exist on my machine. So the 404 makes since. 

Looked around and found the service worker file here: `/usr/lib/code-server/out/browser/serviceWorker.js`.  Copying the contents of this file to `usr/lib/code-server/dist/serviceWorker.js` fixes the 404 error and allows me to install the PWA.

**Reply by jsjoeio** [reply]
Hooray! Thanks for adding those details @HunterHeston - we goofed when we updated the serviceWorker path and forgot to change that from `/dist` to `/out/browser`. That was fixed here: https://github.com/cdr/code-server/pull/3829

Glad you got it working!
